text/0033-private-testnet-guide.md

```diff
@@ -0,0 +1,418 @@
+```sh
+docker build -t tx-mining-service .
+
+docker run -it --network="host" tx-mining-service http://127.0.0.1:8083 --address <address>--api-port 8100 --stratum-port 8101
+```
+
+When running the wallets, we will configure them to use this service for transaction mining.
+
+<!-- # Running our wallets
+We will run 2 wallet-headles to demonstrate the wallet functionality. They are basically wallets that run as a HTTP api and have most of the same functions as the desktop or mobile wallets.
+
+Each wallet-headless will be connected to one of our fullnodes dedicated to wallets (`fullnode-wallet-1` and `fullnode-wallet-2`). But more than one wallet can use the same fullnode without problem. -->
+
+We will run one wallet-headless to demonstrate the wallet functionality. It's basically a wallet that run as a HTTP api and have most of the same functions as the desktop or mobile wallets.
+
+The first step is to clone the repo https://github.com/HathorNetwork/hathor-wallet-headless
+
+## Configuring the Tx Mining Service
+
+We will need to configure our wallets to point to our tx-mining-service.
+
+Due to a still to be fixed issue in the hathor-wallet-headless, we have one extra step to be able to configure our tx-mining-service.
+
+First, install the dependencies:
+
+```sh
+npm install
+```
+
+Then, run the following command:
+```sh
+sed -i 's/https:\/\/txmining.mainnet.hathor.network/http:\/\/localhost:8100/g' node_modules/@hathor/wallet-lib/lib/constants.js
+sed -i 's/https:\/\/txmining.testnet.hathor.network/http:\/\/localhost:8100/g' node_modules/@hathor/wallet-lib/lib/constants.js
+```
+
+With this, we are ready to run our wallets.
+
+## Running the wallet
+
+We will run the wallet using the seed we generated at the beginning of the guide. This seed should have some balance, since we configured it as the destination for mining rewards.
+
+Create the config file, replacing `<seed>` with the seed from the beginning of the guide:
+```sh
+cat <<EOF > src/config.js
+module.exports = {
+  http_bind_address: 'localhost',
+  http_port: 8000,
+  network: 'testnet',
+  server: 'http://localhost:8081/v1a/',
+  seeds: {
+      default: '<seed>',
+  },
+  consoleLevel: null,
+  tokenUid: '',
+  gapLimit: null,
+  connectionTimeout: null,
+  allowPassphrase: false,
+  confirmFirstAddress: false,
+};
+EOF
+```
+
+Start the server:
+```sh
+npm start
+```
+
+Start the wallet:
+```sh
+curl -X POST --data "wallet-id=wallet1" --data "seedKey=default" http://localhost:8000/start
+```
+
+Check the wallet balance:
+```sh
+curl -H "X-Wallet-Id: wallet1" http://localhost:8000/wallet/balance
+```
+
+Get address to send transaction:
+```sh
+curl -H "X-Wallet-Id: wallet1" http://localhost:8000/wallet/address
+```
+
+Send a transaction, replacing the address from the last command output:
+
+> TODO: Getting a timeout in TxMiningService when trying to mine the transaction
```

Comment:
I think I need help with this.

TxMiningService is not able to mine the transactions for some reason, they keep timing out.

I would need to debug this to findout what's happening, if nobody has any idea what this could be.